packages/lesmis-server/src/routes/Route.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+import { PermissionsError } from '@tupaia/utils';
+import { Route as BaseRoute } from '@tupaia/tupaia-server';
+
+import { LesmisSessionType } from '../models';
+import { MeditrakConnection, ReportConnection } from '../connections';
+import { LESMIS_PERMISSION_GROUP } from '../constants';
+
+export class Route extends BaseRoute {
+  meditrakConnection?: MeditrakConnection;
+
+  reportConnection?: ReportConnection;
+
+  async handle() {
+    // All routes will be wrapped with an error catcher that simply passes the error to the next()
+    // function, causing error handling middleware to be fired. Otherwise, async errors will be
+    // swallowed.
+    try {
+      const session = await this.verifyAuth();
+      this.meditrakConnection = new MeditrakConnection(session);
+      this.reportConnection = new ReportConnection(session);
```

Comment:
I would keep just these in the lesmis-server `Route` class, and have the parent include `handle()` with a call to `this.setupConnections` (another abstract method that can be implemented at this level)